java-squid/src/main/java/org/sonar/java/se/symbolicvalues/BinarySymbolicValue.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.se.symbolicvalues;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import org.sonar.java.se.ConstraintManager.BooleanConstraint;
+import org.sonar.java.se.ObjectConstraint;
+import org.sonar.java.se.ProgramState;
+
+import javax.annotation.CheckForNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class BinarySymbolicValue extends SymbolicValue {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Override this superclass' "equals" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2160)